sormas-backend/src/main/java/de/symeda/sormas/backend/event/eventimport/EventImportFacadeEjb.java

```diff
@@ -0,0 +1,615 @@
+							if (infrastructureData.getElement1() != null) {
+								throw new ImportErrorException(
+									I18nProperties.getValidationError(
+										Validations.importEntryDoesNotExistDbOrCommunity,
+										entry,
+										buildEntityProperty(entryHeaderPath)));
+							} else {
+								throw new ImportErrorException(
+									I18nProperties.getValidationError(
+										Validations.importEntryDoesNotExistDbOrDistrict,
+										entry,
+										buildEntityProperty(entryHeaderPath)));
+							}
+						} else if (facilities.size() > 1 && infrastructureData.getElement1() == null) {
+							throw new ImportErrorException(
+								I18nProperties
+									.getValidationError(Validations.importFacilityNotUniqueInDistrict, entry, buildEntityProperty(entryHeaderPath)));
+						} else if (facilities.size() > 1 && infrastructureData.getElement1() != null) {
+							throw new ImportErrorException(
+								I18nProperties
+									.getValidationError(Validations.importFacilityNotUniqueInCommunity, entry, buildEntityProperty(entryHeaderPath)));
+						} else {
+							pd.getWriteMethod().invoke(currentElement, facilities.get(0));
+						}
+					} else {
+						throw new UnsupportedOperationException(
+							I18nProperties.getValidationError(Validations.importEventsPropertyTypeNotAllowed, propertyType.getName()));
+					}
+				}
+			} catch (IntrospectionException e) {
+				throw new InvalidColumnException(buildEntityProperty(entryHeaderPath));
+			} catch (InvocationTargetException | IllegalAccessException e) {
+				throw new ImportErrorException(
+					I18nProperties.getValidationError(Validations.importErrorInColumn, buildEntityProperty(entryHeaderPath)));
+			} catch (IllegalArgumentException e) {
+				throw new ImportErrorException(entry, buildEntityProperty(entryHeaderPath));
+			} catch (ParseException e) {
+				throw new ImportErrorException(
+					I18nProperties.getValidationError(Validations.importInvalidDate, buildEntityProperty(entryHeaderPath)));
+			} catch (ImportErrorException e) {
+				throw e;
+			} catch (Exception e) {
+				LOGGER.error("Unexpected error when trying to import a case: " + e.getMessage());
+				throw new ImportErrorException(I18nProperties.getValidationError(Validations.importCasesUnexpectedError));
+			}
+		}
+	}
+
+	protected FacilityType getTypeOfFacility(String propertyName, Object currentElement)
+		throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+		String typeProperty;
+		if (LocationDto.class.equals(currentElement.getClass()) && LocationDto.FACILITY.equals(propertyName)) {
+			typeProperty = LocationDto.FACILITY_TYPE;
+		} else {
+			typeProperty = propertyName + "Type";
+		}
+		PropertyDescriptor pd = new PropertyDescriptor(typeProperty, currentElement.getClass());
+		return (FacilityType) pd.getReadMethod().invoke(currentElement);
+	}
+
+	protected boolean executeDefaultInvokings(PropertyDescriptor pd, Object element, String entry, String[] entryHeaderPath)
```

Comment:
Since this is probably an exact copy of the same method in CaseImportFacadeEjb, I'd like you to move the code to ImportFacadeEjb and use it both here and in CaseImportFacadeEjb to reduce code duplication